cats/src/io/github/iltotore/iron/cats.scala

```diff
@@ -164,15 +163,14 @@ object cats extends IronCatsInstances:
      * @return a [[Valid]] containing this value as [[IronType]] or an [[Invalid]] containing a [[NonEmptyList]] of error messages.
      * @see [[validated]], [[validatedNec]].
      */
-    inline def validatedNel(value: A)(using inline c: Constraint[A, C]): ValidatedNel[String, T] =
-      value.refineValidatedNel[C].map(_.asInstanceOf[T])
+    inline def validatedNel(value: A): ValidatedNel[String, T] = ops.either(value).toValidatedNel
```

Comment:
(Very) slightly inefficient to convert to Either first, but it allows us to keep things easy to reason with and extend. 

Since the primitives are already boxed and the conversion is encapsulated within a single method there should be extremely minimal impacts to runtime performance & GC

Reply:
True. Another alternartive which is as readable as this one in my opinion is to use an if/else